File: bannerlayout/src/main/java/com/wkz/bannerlayout/annotation/PageTransformerMode.java
package com.wkz.bannerlayout.annotation;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@IntDef({
        PageTransformerMode.ANIMATION_ACCORDION,
        PageTransformerMode.ANIMATION_BACKGROUND,
        PageTransformerMode.ANIMATION_CUBE_IN,
        PageTransformerMode.ANIMATION_CUBE_OUT,
        PageTransformerMode.ANIMATION_DEFAULT,
        PageTransformerMode.ANIMATION_DEPTH_PAGE,
        PageTransformerMode.ANIMATION_DRAWER,
        PageTransformerMode.ANIMATION_FLIPHORIZONTAL,
        PageTransformerMode.ANIMATION_FLIPVERTICAL,
        PageTransformerMode.ANIMATION_FOREGROUND,
        PageTransformerMode.ANIMATION_ROTATEDOWN,
        PageTransformerMode.ANIMATION_ROTATEUP,
        PageTransformerMode.ANIMATION_SCALEINOUT,
        PageTransformerMode.ANIMATION_STACK,
        PageTransformerMode.ANIMATION_TABLET,
        PageTransformerMode.ANIMATION_VERTICAL,
        PageTransformerMode.ANIMATION_ZOOMIN,
        PageTransformerMode.ANIMATION_ZOOMOUTPAGE,
        PageTransformerMode.ANIMATION_ZOOMOUTSLIDE,
        PageTransformerMode.ANIMATION_ZOOMOUT
})
@Retention(RetentionPolicy.CLASS)
public @interface PageTransformerMode {
     int ANIMATION_ACCORDION = 0;
     int ANIMATION_BACKGROUND = 1;
     int ANIMATION_CUBE_IN = 2;
     int ANIMATION_CUBE_OUT = 3;
     int ANIMATION_DEFAULT = 4;
     int ANIMATION_DEPTH_PAGE = 5;
     int ANIMATION_FLIPHORIZONTAL = 6;
     int ANIMATION_FLIPVERTICAL = 7;
     int ANIMATION_FOREGROUND = 8;
     int ANIMATION_ROTATEDOWN = 9;
     int ANIMATION_ROTATEUP = 10;
     int ANIMATION_SCALEINOUT = 11;
     int ANIMATION_STACK = 12;
     int ANIMATION_TABLET = 13;
     int ANIMATION_VERTICAL = 14;
     int ANIMATION_ZOOMIN = 15;
     int ANIMATION_ZOOMOUTPAGE = 16;
     int ANIMATION_ZOOMOUTSLIDE = 17;
     int ANIMATION_ZOOMOUT = 18;
     int ANIMATION_DRAWER = 19;
}

File: bannerlayout/src/main/java/com/fphoenixcorneae/bannerlayout/annotation/PageTransformerMode.kt
package com.fphoenixcorneae.bannerlayout.annotation

import androidx.annotation.IntDef

@IntDef(
        PageTransformerMode.ANIMATION_ACCORDION,
        PageTransformerMode.ANIMATION_BACKGROUND,
        PageTransformerMode.ANIMATION_CUBE_IN,
        PageTransformerMode.ANIMATION_CUBE_OUT,
        PageTransformerMode.ANIMATION_DEFAULT,
        PageTransformerMode.ANIMATION_DEPTH_PAGE,
        PageTransformerMode.ANIMATION_DRAWER,
        PageTransformerMode.ANIMATION_FADE_OUT_FADE_IN,
        PageTransformerMode.ANIMATION_FLIP_HORIZONTAL,
        PageTransformerMode.ANIMATION_FLIP_VERTICAL,
        PageTransformerMode.ANIMATION_FOREGROUND,
        PageTransformerMode.ANIMATION_ROTATE_DOWN,
        PageTransformerMode.ANIMATION_ROTATE_UP,
        PageTransformerMode.ANIMATION_SCALE_IN_OUT,
        PageTransformerMode.ANIMATION_STACK,
        PageTransformerMode.ANIMATION_TABLET,
        PageTransformerMode.ANIMATION_VERTICAL,
        PageTransformerMode.ANIMATION_ZOOM_IN,
        PageTransformerMode.ANIMATION_ZOOM_OUT_PAGE,
        PageTransformerMode.ANIMATION_ZOOM_OUT_SLIDE,
        PageTransformerMode.ANIMATION_ZOOM_OUT
)
@kotlin.annotation.Retention(AnnotationRetention.BINARY)
annotation class PageTransformerMode {
    companion object {
        const val ANIMATION_ACCORDION = 0
        const val ANIMATION_BACKGROUND = 1
        const val ANIMATION_CUBE_IN = 2
        const val ANIMATION_CUBE_OUT = 3
        const val ANIMATION_DEFAULT = 4
        const val ANIMATION_DEPTH_PAGE = 5
        const val ANIMATION_DRAWER = 6
        const val ANIMATION_FADE_OUT_FADE_IN = 7
        const val ANIMATION_FLIP_HORIZONTAL = 8
        const val ANIMATION_FLIP_VERTICAL = 9
        const val ANIMATION_FOREGROUND = 10
        const val ANIMATION_ROTATE_DOWN = 11
        const val ANIMATION_ROTATE_UP = 12
        const val ANIMATION_SCALE_IN_OUT = 13
        const val ANIMATION_STACK = 14
        const val ANIMATION_TABLET = 15
        const val ANIMATION_VERTICAL = 16
        const val ANIMATION_ZOOM_IN = 17
        const val ANIMATION_ZOOM_OUT_PAGE = 18
        const val ANIMATION_ZOOM_OUT_SLIDE = 19
        const val ANIMATION_ZOOM_OUT = 20
    }
}